azavea/raster-foundry
# Intercom integration

## Overview

This PR leverages Intercom to inject a customer support widget into the front-end.

With the new `intercomService` any aspect of the application can interact with Intercom. On login, the user profile is used to boot the widget.

The `bootWithUser` method takes a user profile (which can be obtained by using the authService's `profile` function) and boots the widget. We use this method on state changes and after authentication. Re-booting on state changes is the recommended method of keeping accurate activity records.

The `shutdown` method remove the widget from the app. We use this whenever the user explicitly logs out or is logged out by the app due to end of session or other factors.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

<img width="526" alt="screen shot 2017-03-28 at 11 58 04 am" src="https://cloud.githubusercontent.com/assets/2442245/24415648/2590cbb8-13b0-11e7-9d82-f6748ebe8db8.png">
<img width="491" alt="screen shot 2017-03-28 at 11 57 55 am" src="https://cloud.githubusercontent.com/assets/2442245/24415649/2596214e-13b0-11e7-9eff-0102978ed3c7.png">


### Notes

You will need to grab the `.env` to make sure you have the test intercom app id

There will be an accompanying PR in the deployment repo to handle providing the app id in the staging and production environments.

## Testing Instructions

 * Make sure you are logged out - the intercom widget should not be visible when the user is logged out
 * Log in, see that the intercom widget is present (but not expanded)
 * Interact with the widget, test that state is logically preserved (stays open on route changes, etc)
 * Clear the application storage and click a link in the app that will force a logout, ensure that even on a forced logout the widget doesn't show

Connects #1299
